### PR TITLE
Bug 1561349 - UI polish for What's New messages

### PR DESCRIPTION
**Pull request by rlr**
This requires the latest in https://phabricator.services.mozilla.com/D37320

Screen Shot 2019-07-09 at 4 22 01 PM

Some notes from discussion with Aaron:
* When user clicks on a CTA, the panel will go away... At least for now.
* We might need to tweak the height of the panel depending on the final content.

### CODE REVIEW

File: lib/ToolbarPanelHub.jsm [hunk 1]
```diff
@@ -78,43 +78,44 @@ class _ToolbarPanelHub {
     const container = doc.getElementById(containerId);
 
     if (messages && !container.querySelector(".whatsNew-message")) {
+      let previousDate = 0;
```
Comment (piatra) [inline lib/ToolbarPanelHub.jsm]: Should messages be sorted by date to ensure this will work?

File: lib/OnboardingMessageProvider.jsm [hunk 2]
```diff
@@ -534,8 +537,9 @@ const ONBOARDING_MESSAGES = async () => [
       body:
         "The New Enhanced Tracking Protection, gives you the best level of protection and performance. Discover how this version is the safest version of firefox ever made.",
       link_text: "Learn more on our blog",
-      link_url: "https://blog.mozilla.org/",
+      cta_url: "https://blog.mozilla.org/",
     },
+    priority: 1,
```
Comment (piatra) [inline lib/OnboardingMessageProvider.jsm]: Why was this needed?

Comment (rlr) [reply lib/OnboardingMessageProvider.jsm]: oh right. I should've sorted by date instead of adding this 😝

File: lib/ToolbarPanelHub.jsm [hunk 3]
```diff
@@ -126,11 +136,29 @@ class _ToolbarPanelHub {
       // TODO: TELEMETRY
     });
 
-    messageEl.appendChild(dateEl);
-    messageEl.appendChild(wrapperEl);
+    if (content.icon_url) {
```
Comment (piatra) [inline lib/ToolbarPanelHub.jsm]: Can you combine this with the previous `if` statement on line `123`?

File: test/unit/lib/ToolbarPanelHub.test.js [hunk 4]
```diff
@@ -7,7 +7,6 @@ describe("ToolbarPanelHub", () => {
   let sandbox;
   let instance;
   let everyWindowStub;
-  let message;
```
Comment (piatra) [inline test/unit/lib/ToolbarPanelHub.test.js]: Can you please also add a test for checking that the elements are rendered in sorted order?
I would stub `toLocaleDateString` and check it's only called `X` number of times. Or that the content rendered has only `X` `dateEl` elements.

**Comment by rlr** [comment]
correct. All messages will have a CTA URL but not necessarily an explicit link/button visible. And you can click anywhere on any message to trigger it.
